### DISCUSSION TITLE
Dynamic segment

### DISCUSSION THREAD

**Discussion by ptrstpp950** [JanDeDobbeleer/oh-my-posh#772] [Ideas]
I tried to find it in source code, and on previous discussions but without success. Is it somehow possible to display segment dynamic like in oh-my-zsh? So for example when I type `kubectl<SPACE>` a kubectl segment will show. I know that PowerShell somehow allows that, for example with autocomplete.

**Comment by JanDeDobbeleer** [comment]
There's @TravisTX magic tooltips module: https://github.com/TravisTX/MagicTooltips

But maybe there's a way to do this "natively" for oh-my-posh.

**Reply by ptrstpp950** [reply]
Combo works quite well.

**Reply by JanDeDobbeleer** [reply]
I've been trying out a way yesterday and maybe we can actually facilitate this. In the meantime, you're unblocked 😉

**Reply by ptrstpp950** [reply]
If I can help somehow, count me in :)

**Reply by JanDeDobbeleer** [reply]
I actually have something working on Powershell. Need to refactor it a bit and also support zsh/bash but it should be good to go somewhere next week for a beta run.

**Reply by tradiff** [reply]
I'm curious how you're doing it. Registering a hotkey for ` ` (space) like I did, or did you find some other technique?

**Reply by JanDeDobbeleer** [reply]
@TravisTX I looked at your implementation indeed. That's the most straightforward way but I'm worried it can lead to slowness in certain cases.

**Reply by tradiff** [reply]
Yeah, I was also pretty worried about performance.

**Reply by JanDeDobbeleer** [reply]
@TravisTX @ptrstpp950 we have a [beta](https://ohmyposh.dev/docs/beta#tooltips) for this for pwsh and zsh.
